File: keiner_lindarte_aro/instructor_henry/mis_programas/ruby/orientado_a_objetos/4_clase_de_05_agosto/actividad.rb
class Transporte
  attr_accessor :marca
end

class Terrestre < Transporte
  attr_accessor :llantas
end

class Aereo < Transporte
  attr_accessor :motores
end

class Auto < Terrestre
  attr_accessor :aire

  def registro_carro

    if aire == true
      puts "Se ha registrado un carro con la marca #{self.marca}, con #{self.llantas} llantas y el cual tiene aire acondicionado."
    else
      puts "Se ha registrado un carro con la marca #{self.marca}, con #{self.llantas} llantas y el cual no tiene aire acondicionado."
    end

  end

end

class Moto < Terrestre
  attr_accessor :cascos

  def registro_moto

    puts "Se ha registrado una moto con la marca #{self.marca}, con #{self.llantas} llantas y el cual viene con #{self.cascos} cascos."

  end

end

class Avion < Aereo
  def self.manual

    puts "Es manual"

  end

  def registro_avion
  
    puts "Se ha registrado un avion con la marca #{self.marca}, con #{self.motores} motores y el avion "
    Avion.manual

  end

end

carro = Auto.new
carro.marca = "Toyota"
carro.llantas = 4
carro.aire = true
carro.registro_carro()

moto = Moto.new
moto.marca = "boxer"
moto.llantas = 2
moto.cascos = 2
moto.registro_moto()

avion = Avion.new
avion.marca = "boeing"
avion.motores = 2
avion.registro_avion()

File: keiner_lindarte_aro/instructor_henry/mis_programas/dart/orientado_a_objetos/4_clase_de_05_agosto/actividad.dart
class Transporte{
  String? marca;
}

class Terrestre extends Transporte {
  int? llantas;
}

class Aereo extends Transporte {
  int? motores;
}

class Auto extends Terrestre {
  
  bool? aire;

  void registro_carro(){

      if (aire == true){
    print("Se ha registrado un carro con la marca ${this.marca}, con ${this.llantas} llantas y el cual tiene aire acondicionado.");
    } else {
    print("Se ha registrado un carro con la marca ${this.marca}, con ${this.llantas} llantas y el cual no tiene aire acondicionado.");
    }

  }


}

class Moto extends Terrestre {
  int? cascos;

  void registro_moto(){

    print("Se ha registrado una moto con la marca ${this.marca}, con ${this.llantas} llantas y el cual viene con ${this.cascos} cascos.");


  }


}

class Avion extends Aereo {
  static void manual(){
    print("Es manual");
  }

  void registro_avion(){

    print("Se ha registrado un avion con la marca ${this.marca}, con ${this.motores} motores y el avion "); Avion.manual();


  }
}

void main (){

  Auto carro = new Auto();
  carro.marca = "Toyota";
  carro.llantas = 4;
  carro.aire = true;
  carro.registro_carro();


  Moto moto = new Moto();
  moto.marca = "boxer";
  moto.llantas = 2;
  moto.cascos = 2;
  moto.registro_moto();

  Avion avion = new Avion();
  avion.marca = "boeing";
  avion.motores = 2;
  avion.registro_avion();

}